api-operator/pkg/registry/http.go

```diff
@@ -18,28 +18,29 @@ package registry
 
 import (
 	"fmt"
-	"github.com/wso2/k8s-api-operator/api-operator/pkg/registry/utils"
 	"strings"
+
+	"github.com/wso2/k8s-api-operator/api-operator/pkg/registry/utils"
 )
 
 const HTTP Type = "HTTP"
 
-// Copy of Docker Hub configs as HTTP private registry configs
-var httpReg = *dockerHub
-
+// getHttpRegConfigFunc copies  Docker Hub configs as HTTP private registry configs
 func getHttpRegConfigFunc(repoName string, imgName string, tag string) *Config {
+	var httpReg = getDockerHubConfigFunc(repoName, imgName, tag)
+
 	httpReg.ImagePath = fmt.Sprintf("%s/%s:%s", repoName, imgName, tag)
 	// Add --insecure arg
 	// This will not effect Docker Hub configs since this is a copy of it
 	httpReg.Args = []string{"--insecure"}
 	// replace https with http and use default function
-	httpReg.IsImageExist = func(config *Config, auth utils.RegAuth, image string, tag string) (b bool, err error) {
+	httpReg.IsImageExist = func(config *Config, auth utils.RegAuth, imageRepository string, imageName string, tag string) (b bool, err error) {
 		auth.RegistryUrl = strings.Replace(auth.RegistryUrl, "https://", "http://", 1)
 		logger.Info("Checking for image in HTTP registry", "Registry URL", auth.RegistryUrl)
-		return utils.IsImageExists(auth, getImageWithoutReg(image), tag)
+		return utils.IsImageExists(auth, "", imageName, tag)
```

Comment:
In `docker-registry-config` we define `registryType` and  `repositoryName`.
Example: for private HTTP registry
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: docker-registry-config
  namespace: wso2-system
data:
  registryType: HTTP
  repositoryName: 192.168.8.132:5000/apis-test
```

We assign the API name for the image name. So if we create an API named `pets` (version v1), the image is `192.168.8.132:5000/apis-test/pets:v1`

Hence:
- Repository: 192.168.8.132:5000/apis-test
- Image: pets
- Tag: v1

In this function (`utils.IsImageExists`), we have referred `apis-test/pets` as imageName in the current code (using the function `getImageWithoutReg`). But your changes referring it to `pets` (I think we have a confusion with current variable names). So the Docker registry API request is `"http://192.168.8.132:5000/v2/pets/tags/list"` instead of `"http://192.168.8.132:5000/v2/apis-test/pets/tags/list"` which should be corrected.